src/views/Collectibles/components/ClaimNftModal.tsx

```diff
@@ -30,14 +30,16 @@ const ClaimNftModal: React.FC<ClaimNftModalProps> = ({ nft, onSuccess, onClaim,
   const { toastError, toastSuccess } = useToast()
 
   const handleConfirm = async () => {
-    const tx = await onClaim()
     setIsConfirming(true)
-    const receipt = await tx.wait()
-    if (receipt.status) {
-      toastSuccess(t('Successfully claimed!'))
-      onDismiss()
-      onSuccess()
-    } else {
```

Comment:
In this previous pattern the `else` condition isn't hit when the tx fails, so the fail state functions never trigger.